hudi-common/src/main/java/org/apache/hudi/common/table/read/HoodieBaseFileGroupRecordBuffer.java

```diff
@@ -306,14 +306,16 @@ protected Option<DeleteRecord> doProcessNextDeletedRecord(DeleteRecord deleteRec
           if (isDeleteRecordWithNaturalOrder(existingRecordMetadataPair.getLeft(), existingOrderingVal)) {
             return Option.empty();
           }
-          Comparable deleteOrderingVal = deleteRecord.getOrderingValue();
-          // Checks the ordering value does not equal to 0
-          // because we use 0 as the default value which means natural order
-          boolean chooseExisting = !deleteOrderingVal.equals(0)
```

Comment:
We can leverage existing logic by simply setting the ordering value to 0 if we want natural order, without `DELETE_FOUND_WITHOUT_ORDERING_VALUE`?